### PR TITLE
clippy: Fix large size difference between variants of `ScriptToCompositorMsg`

### PR DESCRIPTION
**Pull request by chickenleaf**
**file**: components/shared/webrender/lib.rs

**PR Description**
**Issue**: Clippy warned about `ScriptToCompositorMsg` having a big size difference between variants.

**Change**:  
        1. Made `display_list_info` a `Box<CompositorDisplayListInfo>` in the `SendDisplayList` variant.
        2. Adjusted how `SendDisplayList` is constructed.

**Impact**: One extra heap allocation, but improves performance when using the enum. Functionality stays the same.

---

- [X] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [X] These changes are part of https://github.com/servo/servo/issues/31500

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because they do not touch functionality.

### CODE REVIEW

File: components/shared/webrender/lib.rs [hunk 1]
```diff
@@ -337,13 +337,12 @@ impl WebRenderScriptApi {
...
         let (display_list_sender, display_list_receiver) = ipc::bytes_channel().unwrap();
         if let Err(e) = self.0.send(ScriptToCompositorMsg::SendDisplayList {
-            display_list_info,
+            display_list_info: Box::new(display_list_info),
             display_list_descriptor,
             display_list_receiver,
         }) {
             warn!("Error sending display list: {}", e);
         }
-
```
Comment (mrobinson) [inline components/shared/webrender/lib.rs:346]: Can you please preserve this empty line as it's unrelated to your change?

**Review by mrobinson — CHANGES_REQUESTED** [review]

**Comment by mrobinson** [comment]
Nice work! You will need to sign your commit (see https://github.com/servo/servo/pull/33690/checks?check_run_id=31183256151) and I have left one comment about an unnecessary blank line deletion. Thank you.

**Comment by chickenleaf** [comment]
@mrobinson  yes thank you!! shall sign and add back the empty line right away

File: components/shared/webrender/lib.rs (hunk 1 above)
Comment (mrobinson) [reply components/shared/webrender/lib.rs:346]: You have added the empty line back, but with trailing white space.

File: components/shared/webrender/lib.rs [hunk 2]
```diff
@@ -337,13 +337,13 @@ impl WebRenderScriptApi {
...
+
```
Comment (mrobinson) [inline components/shared/webrender/lib.rs:346]:
```suggestion

```

**Review by mrobinson — APPROVED** [review]

**Pull request merged**